gcc/cp/contracts.cc

```diff
@@ -1717,6 +1891,69 @@ build_contract_violation (tree contract, contract_continuation cmode)
   return ctor;
 }
 
+static int
+get_contract_assertion_kind(tree contract)
+{
+  switch (TREE_CODE (contract))
+  {
+    case ASSERTION_STMT:	return CAK_ASSERT;
+    case PRECONDITION_STMT:	return CAK_PRE;
+    case POSTCONDITION_STMT:	return CAK_POST;
+    default:
+      error ("Invalid contract kind");
```

Comment:
is this intended to be a user-facing error? (if so, then it probably should be `error_at` and use the source location that triggered this action.